Deeplink: Clear the URLS after processing

estib-vega opened the pull request:
Clear the deeplink URLs after emission, that way the application doesn’t navigate back after reload

estib-vega commented:
hmmm, this doesn't seem to work.
We need to know whether the application already processed the deeplink, so that on app reload we don't redirect them again.
This doesn't seem to be cleared (or clearable) through the Tauri APIs...?

Byron commented:
That's strange and has the makeup of a bug.
I never read the tauri docs on deep linking though, so maybe there is something special to be done.
There is a chance, of course, that it's half-baked as well.

Would a front-end based fix be tenable, so it remembers that it followed the link already? Or is it too baked in and the URL jump is triggered by Tauri?

estib-vega commented:
> That's strange and has the makeup of a bug. I never read the tauri docs on deep linking though, so maybe there is something special to be done. There is a chance, of course, that it's half-baked as well.
> 
https://v2.tauri.app/plugin/deep-linking/

Nothing I can immediately find. It's weird and might indeed be a bug.

> Would a front-end based fix be tenable, so it remembers that it followed the link already? Or is it too baked in and the URL jump is triggered by Tauri?

https://github.com/tauri-apps/plugins-workspace/issues/1879#issuecomment-2458108929

I see some mention on workarounds from the FE that I could still try out. Man, it sucks that this behavior seems to be known but not documented

Byron commented:
Thanks for the links! I couldn't make much of the issue, but at least could take a look at the docs for the first time.

They don't seem to mention how to remove the deeplink URLs or clear its state, which I understand is the problem here.
And… I don't see know how `getCurrent` can be tuned to not return the same URLs each time.
My hope would be that it pushes onto the list, so each deep link is added to the end of it. Then it would be possible to keep track of the last seen list length in the Window state, non-persisting.
But I am also sure that's not how it works, isn't it?

What is most annoying about this certainly is the debugging workflow, as it needs a real nightly build each time. Doing this locally *should* be OK by now as it builds relatively quickly. But 2m is still not fast, but faster than waiting on CI.

estib-vega committed:
Deeplink: Add timestamp for to the deeplink

In order to determine whether a deeplink has already been processed by the frontend, add a timestamp and persist it.
That way we’ll know whether the link was previously processed.

Byron commented:
That's a great idea! And one that definitely works 🎉!

krlvi commented:
this timestamp approach is a really cool solution! 🎉
